autoload/fzf/vim.vim

```diff
@@ -28,22 +28,54 @@ set cpo&vim
 " Common
 " ------------------------------------------------------------------
 
+function! s:get_if_linux_like_bash()
+  if s:is_win
+    let s:is_defined_win_bash = get(g:, 'fzf_preview_win_bash', '') != ''
+    if s:is_defined_win_bash
+      let bash = g:fzf_preview_win_bash
+      let s:defined_win_bash = split(system('for %A in ("'.g:fzf_preview_win_bash.'") do @echo %~sA'), "\n")[0]
+    else
+      let bash = exepath('bash')
+    endif
+
+    if exists('g:fzf_preview_win_bash_linux_like')
+      return g:fzf_preview_win_bash_linux_like
+    endif
+
+    for _re in s:known_wsl_like_bash
+      if match(bash, _re) != -1
+        return 1
+      endif
+    endfor
+  endif
+  return 0
+endfunction
+
+function! s:set_preview_bash()
+  if s:is_win && s:is_defined_win_bash
+    let s:preview_bash = s:defined_win_bash
+  else
+    let s:preview_bash = 'bash'
+  endif
+endfunction
+
+
 let s:min_version = '0.23.0'
+let s:known_wsl_like_bash = ['\cwindows[/\\]system32[/\\]bash\(.exe\)\=$']
 let s:is_win = has('win32') || has('win64')
-let s:is_wsl_bash = s:is_win && (exepath('bash') =~? 'Windows[/\\]system32[/\\]bash.exe$')
+let s:is_wsl_bash = s:get_if_linux_like_bash()
 let s:layout_keys = ['window', 'up', 'down', 'left', 'right']
 let s:bin_dir = expand('<sfile>:p:h:h:h').'/bin/'
 let s:bin = {
 \ 'preview': s:bin_dir.'preview.sh',
 \ 'tags':    s:bin_dir.'tags.pl' }
 let s:TYPE = {'dict': type({}), 'funcref': type(function('call')), 'string': type(''), 'list': type([])}
+call s:set_preview_bash()
 if s:is_win
-  if has('nvim')
+  if !s:is_wsl_bash
```

Comment:
So we no longer need to differentiate between neovim and vim?

Reply:
Yes through my tests, we just need to handle the path correctly.